guides/plugins/plugins/administration/add-rule-assignment-configuration.md

```diff
@@ -0,0 +1,290 @@
+                const api = Object.assign({}, Context.api);
+                api.inheritance = true;
+
+                return api;
+            },
+            detailRoute: 'sw.product.detail.base',
+            gridColumns: [
+                {
+                    property: 'name',
+                    label: 'Name',
+                    rawData: true,
+                    sortable: true,
+                    routerLink: 'sw.product.detail.prices',
+                    allowEdit: false,
+                },
+            ],
+            deleteContext: {
+                type: 'many-to-many',
+                entity: 'product',
+                column: 'extensions.swagDynamicAccessRules',
+            },
+            addContext: {
+                type: 'many-to-many',
+                entity: 'swag_dynamic_access_product_rule',
+                column: 'productId',
+                searchColumn: 'name',
+                criteria: () => {
+                    const criteria = new Criteria();
+                    criteria.addFilter(
+                            Criteria.not('AND', [Criteria.equals('swagDynamicAccessRules.id', this.rule.id)]),
+                    );
+                    criteria.addAssociation('options.group');
+
+                    return criteria;
+                },
+                gridColumns: [
+                    {
+                        property: 'name',
+                        label: 'Name',
+                        rawData: true,
+                        sortable: true,
+                        allowEdit: false,
+                    },
+                    {
+                        property: 'productNumber',
+                        label: 'Product number',
+                        rawData: true,
+                        sortable: true,
+                        allowEdit: false,
+                    },
+                    {
+                        property: 'manufacturer.name',
+                        label: 'Manufacturer',
+                        rawData: true,
+                        sortable: true,
+                        allowEdit: false,
+                    },
+                    {
+                        property: 'active',
+                        label: 'Active',
+                        rawData: true,
+                        sortable: true,
+                        allowEdit: false,
+                    },
+                    {
+                        property: 'stock',
+                        label: 'Stock',
+                        rawData: true,
+                        sortable: true,
+                        allowEdit: false,
+                    },
+                    {
+                        property: 'availableStock',
+                        label: 'Available',
+                        rawData: true,
+                        sortable: true,
+                        allowEdit: false,
+                    },
+                ],
+            },
+        },
+    ];
+}
```

Comment:
Would it be possible to shorten this in any way?

Reply:
I will only set one column definition and remove the rest.